src/components/PopupWithForm/PopupWithForm.js

```diff
@@ -5,69 +5,92 @@ import "./__input/popup-with-form__input.css";
 
 import Popup from "../Popup/Popup.js";
 
-function PopupWithForm({ isOpen, onClose, onRegister }) {
+function PopupWithForm({
+  isOpen,
+  onClose,
+  onRegister,
+  handleChange,
+  values,
+  errors,
+  isValid,
+}) {
   return (
     <Popup isOpen={isOpen} onClose={onClose}>
       <form className="popup-with-form">
         <h2 className="popup-with-form__title">Sign in</h2>
 
-        <label for="email-input" className="popup-with-form__label">
-          Email
-        </label>
+        <label className="popup-with-form__label">Email</label>
         <input
           type="email"
           name="email"
           id="email-input"
           placeholder="Enter email"
           className="popup-with-form__input"
+          noValidate
           required
+          onChange={handleChange}
+          value={values.email || ""}
         />
         <div className="popup-with-form__error-message">
-          Invalid email address
+          {`${errors.email ? errors.email : ""}`}
         </div>
 
-        <label for="password-input" className="popup-with-form__label">
-          Password
-        </label>
+        <label className="popup-with-form__label">Password</label>
         <input
           type="password"
           name="password"
           id="password-input"
           placeholder="Enter password"
           className="popup-with-form__input"
+          noValidate
           required
+          onChange={handleChange}
+          value={values.password || ""}
+          minLength="8"
+          maxLength="30"
         />
-        <div className="popup-with-form__error-message">Invalid password</div>
+        <div className="popup-with-form__error-message">
+          {`${errors.password ? errors.password : ""}`}
+        </div>
 
-        <label for="username-input" className="popup-with-form__label">
-          Username
-        </label>
+        <label className="popup-with-form__label">Username</label>
         <input
           type="text"
           name="username"
           id="username-input"
           placeholder="Enter your username"
           className="popup-with-form__input"
+          minLength="2"
+          maxLength="30"
+          noValidate
           required
+          onChange={handleChange}
+          value={values.username || ""}
         />
-        <div className="popup-with-form__error-message">Invalid username</div>
+        <div className="popup-with-form__error-message">
+          {`${errors.username ? errors.username : ""}`}
+        </div>
 
         <div className="popup-with-form__error-message popup-with-form__error-message_bad-email">
           This email is not available
         </div>
         <button
           type="submit"
-          className="popup-with-form__send-button"
+          className={`popup-with-form__send-button ${
+            !isValid ? "popup-with-form__send-button_disabled" : ""
+          }`}
           onClick={(event) => {
```

Comment:
Submit handlers are added only to tags `form` with event `submit` rather than to submit buttons with event `click` because it also automatically handles `Enter` presses